src/metabase/lib/metadata.cljc

```diff
@@ -220,7 +220,8 @@
   If metadata for the `:source-table` or `:source-card` can be found, then the query is editable."
   [query :- ::lib.schema/query]
   (let [{:keys [source-table source-card] :as stage0} (lib.util/query-stage query 0)]
-    (boolean (and (database query)
+    (boolean (and (when-let [{:keys [id]} (database query)]
+                    (= (:database query) id))
```

Comment:
It's surprising to me that these would ever differ.

Reply:
In a perfect world, they wouldn't. In the world we inhabit, the metadata is a bit hacky and might close over "The Database" (or its ID) and always return that even when handed a `MetadataProvider` like a query aimed at another database.